src/category_theory/quotient.lean

```diff
@@ -0,0 +1,101 @@
+import category_theory.functor
+import category_theory.eq_to_hom
+
+/-
+  This file defines quotient categories. Given a category and an arbitrary family of relations
+  on its homsets, thought of as identifying some homs, we construct a new category on an
+  identical type of objects, where related homs are identified.
+
+    Note that we do *not* assume that our relation behaves well under composition, or that it is
+  an equivalence relation.
+-/
+
+universes v v₁ u u₁
+
+namespace category_theory
+
+variables
+  {C : Type u} [category.{v} C]
+  (r : Π (a b : C), (a ⟶ b) → (a ⟶ b) → Prop)
+include r
+
+/-- The type of objects in the quotient category --/
+structure quotient := (to_C : C)
+def of_C (a : C) : quotient r := ⟨r, a⟩
+
+namespace quotient
+
+/- Generates the closure of a family of relations wrt composition from left and right
+  If m₁ and m₂ are identified, then fm₁g and fm₂g should also be identified. -/
+inductive ccl {s t : C} : (s ⟶ t) → (s ⟶ t) → Prop
```

Comment:
Meaningful name? Even after reading the comment I can't guess what this is an acronym of.

Also, I wonder if this would be easier with three constructors, to introduce something satisfying `r`, to compose on the left, and to compose on the right. Perhaps the lemmas `comp_left` and `comp_right` would just be the constructors themselves.

Reply:
I'll try to find a better name.

You could have constructors for `comp_*`. This would mean that the constructors for `ccl` take `ccl` as input, and `lift` would require an (easy) induction rather than a simple case analysis.

Reply:
How about `comp_closure`?  By the way, as I see it, you shouldn't need to refer to this relation outside this file, as everything can be done via `sound`, `induction`, and `lift`.

Reply:
`comp_closure` is good. Even though this relation can remain private in the file, this file needs to be comprehensible to future readers. 

I agree your choice of definition is fine, no need to switch.